mycroft/skills/skill_manager.py

```diff
@@ -143,6 +146,9 @@ def __init__(self, bus, watchdog=None):
         self._define_message_bus_events()
         self.daemon = True
 
+        self.lang_code = self.config.get("lang", "en-us")
+        load_languages([self.lang_code, "en-us"])
```

Comment:
How does `load_languages` differ from `set_default_lang`?  Why do we need both?

Reply:
see the changelog of the release that introduced this https://github.com/MycroftAI/lingua-franca/releases/tag/release%2Fv0.3.1

and this PR https://github.com/MycroftAI/lingua-franca/pull/108

Reply:
I just looked and there was no reasoning included.  Just that loading languages must now be done.  So my question remains.  Also, it looks like there is `load_language` and `load_languages`.  If we are only loading one language, why use the plural version?

Reply:
```python
lingua_franca.load_language('en')          # 'primary' lang code (package's terminology)
lingua_franca.load_language('pt-pt')       # full lang code
lingua_franca.load_languages(['en', 'es']) # multiple languages at once, first will be default
lingua_franca.set_default_lang('es')       # self-explanatory
```

`default_lang` is the lang to be used if no `lang` is passed to the methods, `load_language` is what actually loads the lang data into memory, the linked PR goes into detail on the rationale of why they arent just all loaded at once (memory saving and performance reasons)

Reply:
* `set_default_lang` sets the default value of the `lang` parameter in localized functions (anything that accepts that parameter =P)
* `load_language` loads a single language's functions into memory, and expects a string
* `load_languages` loads multiple languages at once, and expects a list, but performs a sanity check so that one should never need to *invoke* them both

In Mycroft's case, many things still crash when English parsers and formatters are unavailable, so when the user's configured language is not English, we need to load both.

Reply:
i suppose set_default_lang could be made to automatically load the lang if needed, there is also a config option to load langs on demand. but thats a lingua_franca issue, in the next LF release we can add that, and when it is bumped again in core remove one of these function calls. in the current version of LF both calls are needed

Reply:
Incidentally, calling `load_language` with no default set will set that language to the default. Similarly, calling `load_languages` with no default set will use the first element of the input.

Put differently, if no default language is set,

* `load_language('en')` will make English the default ('en-us', to be specific, but that's for the next version bump)
* `load_languages(['es', 'en'])` will make *Spanish* the default, because Spanish is the first element in its input.

This is why we don't need to call `set_default_lang` after this `load_languages` call.

Reply:
Ok, but `set_default_lang` is called elsewhere in this PR.  Why even expose `set_default_lang` if you need to call `load_language` anyway and it does the work for you?

Reply:
Drive-by: 

One may wish to change the default language at runtime. Indeed, the three functions were written with Mycroft localization in mind; locales' or entire languages' parsers and formatters can be loaded and unloaded alongside wake words, STT and TTS plugins/models, etc, in one big "change Mycroft language" algorithm.

If I could figure out a way to help users who change to a language they don't speak, and then can't change back, I'd write a skill to do that today, YMMV =P

As to where it's called in this PR, I noticed the one in `reset_converse`, replacing an `active_lang` call. This ensures that the default language didn't somehow get desynced from Mycroft, which is one possible cause of a failure (though unlikely, since the func that would be responsible is wrapped.)

Otherwise, I saw it called in tests. Did you notice it anywhere else?

Reply:
`load_language` doing the work for you is just sugar. If you're just importing LF to use it in a particular language, there's only one workflow, one function call, off you go.

For everything else there's MasterCard.

Reply:
It may also be worth explaining what all this is for:

Probably goes without saying, and just in general, Python doesn't usually release memory. It won't grab more, though, if the heap's big enough for its next assignment or import.

Lingua Franca is small, in terms of what a modern computer can do. However, every time a localized function is added, it grows by the number of supported languages. Every time it acquires a new language, it balloons by the number of localized functions. (This isn't entirely accurate, because the guts and helpers of a localized parser don't necessarily resemble the inside of its English counterpart, but that more or less comes out in the wash.)

That config option, `load_langs_on_demand`, makes all this unnecessary for use cases that don't know what language(s) they're using until the functions are called. But for Mycroft, we noticed that you don't actually need *most* of LF at any given moment. You just need the modules particular to your languages.

So now it's got a cache of function signatures for each loaded language and "top-level" module (`parse` and `format`.) The memory footprint is thus reduced in three ways:

* Localized modules that you are using are imported for introspection, then freed for GC
* Localized modules that you aren't using might as well not be there
* When you actually call a localized function, the function signature cache is consulted, and the appropriate function call is pre-assembled. The localized module is then imported, the function called, and once again the localized module is freed for GC

In percentage terms, this was a radical improvement. `load_langs_on_demand` makes the opposite trade; even lower memory consumption, but far more floppydoos.

Now LF goes fast on electronic vegetables, and fits the larger "this is an English language voice stack for Mycroft" paradigm that will surely define language support.

Reply:
Credit for this goes to three people: Jarbas, for cleaning everything up and standardizing how it's named, which is useful for the localizer-cache thing. Me, for deciding to Try It And See. Ake, for realizing that, no, it *wasn't* an unworkable idea, it just needs to cache function signatures instead of modules. Duh.